File: problem41/problem41.js
/* Problem 41
 * permutationsOfAString: returns all permutations of a string in an array
 * isPrime: returns whether a positive integer is prime
*/

var Combinatorics = require("../../Documents/projet_personnel/informatique/javaScript/Combinatorics.js");
var IntegerArithmetic = require("../../Documents/projet_personnel/informatique/javaScript/IntegerArithmetic.js");
var T = ["123456789", "12345678", "1234567", "123456", "12345", "1234", "123", "12"];
var nbPanPrime = 0; var max = 0;
for (var i = 0; i < T.length; i++) {
	var permutations = Combinatorics.permutationsOfAString(T[i]);
	for (var j = 0; j < permutations.length; j++) {
		var p = parseInt(permutations[j]);
		if (IntegerArithmetic.isPrime(p)) {
			nbPanPrime++;
			if (p > max) max = p;
		}
	}
	if (nbPanPrime > 0) break;
}
console.log("The solution is " + max);

File: problem41/Problem41.java
import java.util.*;
/* Problem 41
 * permutationsOfAString: returns all permutations of a string in an array
 * isPrime: returns whether a positive integer is prime
*/

public class Problem41 {
	public static void main(String[] arg) {
		String[] T = {"123456789", "12345678", "1234567", "123456", "12345", "1234", "123", "12"};
		int nbPanPrime = 0; int max = 0;
		for (int i = 0; i < T.length; i++) {
			String[] permutations = Combinatorics.permutationsOfAString(T[i]);
			for (int j = 0; j < permutations.length; j++) {
				int p = Integer.parseInt(permutations[j]);
				if (IntegerArithmetic.isPrime(p)) {
					nbPanPrime++;
					if (p > max) max = p;
				}
			}
			if (nbPanPrime > 0) break;
		}
		System.out.println("The solution is " + max);
	}
}

